sycl/doc/SYCLGraphUsageGuide.md

```diff
@@ -0,0 +1,493 @@
+# SYCL-Graph Usage Guide and Examples
+
+This document describes recommended usage guidelines for using the
+`sycl_ext_oneapi_graph` extension (referred to as SYCL-Graph in this document)
+as well as provides various example code snippets for various features and usage
+scenarios.
+
+The specification for the `sycl_ext_oneapi_graph` extension can be found
+[here](./extensions/experimental/sycl_ext_oneapi_graph.asciidoc).
+
```

Comment:
We might want to start with a simple example, e.g. Record & Replay that just works, before going into details and talk about things that don't work.

Reply:
Would you suggest moving the entire examples section above these specific usage scenarios?

The intent here is not to provide a beginners guide to record and replay (IMO reading the spec and the examples is best for that), but provide common pitfalls and best practices users may encounter. Something more akin to the C++ core guidelines in specificity. I think we have to assume some familiarity here or it will quickly become cluttered as we add more and more guidelines if we have to give an explanation and examples for each thing from the spec that an example might touch on.